backends/p4tools/p4tools.def

```diff
@@ -241,3 +241,20 @@ public:
           concolicId(concolicId),
           concolicMethodName(methodName) {}
 }
+
+/// An extension of a struct expression. Also tracks validity as an expression.
+class HeaderExpression : StructExpression {
```

Comment:
I am wondering if this could clash with the need for something similar in the core IR (i.e. not just for p4tools). But I guess if that happens this can be moved or changed.

Reply:
Actually, the compiler already has a similar expression in the form of an [InvalidHeader](https://github.com/p4lang/p4c/blob/main/ir/expression.def#L492) and StructExpression (valid header). What we need, which is not yet captures is a HeaderExpression that can model validity symbolically.

I thought about this, but there does not seem to be a use case for this in the general compiler.

Reply:
OK, let's keep it this way. It is unlikely the compiler will need arbitrary expressions as validity bits (unless it starts doing SMT-based optimization :-D).